### PR TITLE
Refactor runOnWhatever to store clk var in e2 context

### LINKED ISSUE #1585: datasignal dsclk returns 1 to all e2 in same tick even if e2 was executed by something else
While debug my os as always i found that dsClk returns 1 to all e2 in same tick even if e2 was executed by something else like input or something else. Here is short instruction of about what i mean:

1. Spawn one e2 with this code:
```lua
@name DS Check 1
@inputs Send SEnt:entity
@persist 
@trigger 

if (~Send & Send) {
    print("DS Check 1")
    dsSendDirect("TEST",SEnt,"Test Message")   
}
```

2. Spawn second e2 with this code:
```lua
@name DS Check 2
@outputs Ent:entity TrigMe

if (first()|duped()) {
    Ent = entity()   
}

if (dsClk()) {
    print("DS Check 2", dsGetString())
    TrigMe = !TrigMe
}
```

3. Spawn third e2 with this code:
```lua
@name DS Check 3
@inputs TrigMe

if (~TrigMe) {
    print("DS Check 3", dsClk(), dsGetString())      
}
```

4. Connect Send input on DS Check 1 to button.
5. Connect SEnt:entity on DS Check 1 to Ent:entity from DS Check 2.
6. Connect TrigMe input on DS Check 3 to TrigMe output from DS Check 2.
7. Press button.

If everything connected right you will get:

```lua
DS Check 1	
DS Check 2	Test Message	
DS Check 3	1	Test Message	
```

So basically on DS Check 3 it reports that it was also triggered by datasignal, but it was triggered by input only! I'm not sure if this is a feature or bug but still reporting about this.

### PR DESCRIPTION
**Pull request by Divran**
Fixes #1585

### CODE REVIEW

File: lua/entities/gmod_wire_expression2/core/files.lua [hunk 1]
```diff
@@ -239,11 +239,11 @@ e2function void runOnFile( active )
 end
 
 e2function number fileClk()
-	return run_on.file.run
+	return self.data.runOnFile and 1 or 0
 end
 
 e2function number fileClk( string filename )
-	return (run_on.file.run == 1 and run_on.file.name == filename) and 1 or 0
+	return (self.data.runOnFile and run_on.file.name == filename) and 1 or 0
```
[the remark this answers is not in the document]
Comment (AbigailBuccaneer) [reply lua/entities/gmod_wire_expression2/core/files.lua]: Ignore that for now - I think this whole file extension might have multiplayer issues due to using the `run_on` global throughout. (Or the extension system works in a surprising way.)

Something to be investigated separately.

**Comment by thegrb93** [comment]
Does this have any benefit? Looks like it just adds the overhead of table indexing to clk variables.

**Comment by bigdogmat** [comment]
See #1586

**Comment by thegrb93** [comment]
Oh ok

**Comment by thegrb93** [comment]
@AlexALX have you tried this? Does it fix the issue

**Comment by thegrb93** [comment]
As long as there's no errors, lgtm

**Comment by thegrb93** [comment]
I can revert or fix it if there's a regression. No idea how to test this

**Comment by CaptainPRICE** [comment]
Ok, let me test... Hoping onto this branch.

**Comment by CaptainPRICE** [comment]
Cool, I have just tested it, and yeah it has been fixed by this PR.

**Comment by CaptainPRICE** [comment]
Still want to test it on your own?
No problem here you go, I made my own advdupe for it:
[wire_issue_1585.txt](https://github.com/wiremod/wire/files/3057147/wire_issue_1585.txt)

Place the text file in `garrysmod/data/advdupe2` folder.
Spawn it using Advanced Duplicator 2, and then use the button.
Then if you see the following print:
`[DS Check 3] #1585 BUG TRIGGERED (NOT FIXED)`

Obviously. It means a bug is not fixed. If you don't see such a print, then the bug has been fixed.

**Comment by thegrb93** [comment]
Allright, thanks for the verification

**Pull request merged**